modules/dashboard/php/endpoints/dashboard.class.inc

```diff
@@ -0,0 +1,570 @@
+        $behaviouralFeedbackData = array();
+
+        $factory = \NDB_Factory::singleton();
+        $DB      = $factory->database();
+        $user    = $factory->user();
+        $config  = $factory->config();
+        $site    = $user->getSiteNames();
+
+        $userID    = $user->getUsername();
+        $lastLogin = $DB->pselectOne(
+            "SELECT MAX(Login_timestamp)
+             FROM user_login_history
+             WHERE Login_timestamp < (SELECT MAX(Login_timestamp)
+                FROM user_login_history
+                WHERE userID=:UserID AND Success='Y')
+             AND userID=:UserID AND Success='Y'",
+            array('UserID' => $userID)
+        );
+
+        $siteID = $user->getCenterIDs();
+        // todo user_site used with myTasks
+        $myTasksData['user_site'] = $siteID;
+
+        // Welcome panel
+        // todo username used with welcome
+        $welcomeData['username'] = $user->getFullname();
+        if ($lastLogin) {
+            // todo lastLogin used with welcome
+            $welcomeData['lastLogin'] = $lastLogin;
+        } else {
+            // todo
+            $welcomeData['lastLogin'] = '...never. Welcome!';
+        }
+        // todo projectDescription used with welcome
+        $welcomeData['projectDescription']
+            = $config->getSetting('projectDescription');
+        $dashboard_links = $config->getExternalLinks('dashboard');
+        if (!empty($dashboard_links)) {
+            foreach ($dashboard_links as $text => $url) {
+                // todo dashboard_links used with welcome
+                $welcomeData['links'][]
+                    = array(
+                        'url'        => $url,
+                        'label'      => $text,
+                        'windowName' => md5($url),
+                    );
+            }
+        }
+
+        $recruitmentTarget = $config->getSetting('recruitmentTarget');
+        $recruitmentData   = $this->createProjectProgressBar(
+            'overall',
+            'Overall Recruitment',
+            intval($recruitmentTarget),
+            $this->getTotalRecruitment(),
+            $recruitmentData
+        );
+
+        $projects = \Utility::getProjectList();
+        foreach ($projects as $projectID => $project) {
+            $projectInfo = $config->getProjectSettings($projectID);
+            if (is_null($projectInfo)) {
+                throw new \LorisException(
+                    'No project settings exist in the Database for ' .
+                    'project ID ' . intval($projectID)
+                );
+            }
+            $recruitmentData = $this->createProjectProgressBar(
+                $projectID,
+                $projectInfo['Name'],
+                intval($projectInfo['recruitmentTarget']),
+                $this->getTotalRecruitmentByProject($projectID),
+                $recruitmentData
+            );
+        }
+
+        // Charts
+        // todo total_scans used with recruitment
```

Comment:
Is this done or not? If not it should be a github issue.

Reply:
@johnsaigle these todo messages shouldn't exist and they're just from the previous author. The code is necessary though.